merkle-tree/concurrent/src/errors.rs

```diff
@@ -65,25 +49,18 @@ impl From<ConcurrentMerkleTreeError> for u32 {
             ConcurrentMerkleTreeError::IntegerOverflow => 2001,
             ConcurrentMerkleTreeError::HeightZero => 2002,
             ConcurrentMerkleTreeError::ChangelogZero => 2003,
-            ConcurrentMerkleTreeError::HeightHigherThanMax => 2004,
-            ConcurrentMerkleTreeError::RootsZero => 2005,
-            ConcurrentMerkleTreeError::RootHigherThanMax => 2006,
-            ConcurrentMerkleTreeError::BytesRead => 2007,
-            ConcurrentMerkleTreeError::TreeFull => 2008,
-            ConcurrentMerkleTreeError::BatchGreaterThanChangelog(_, _) => 2009,
-            ConcurrentMerkleTreeError::InvalidProofLength(_, _) => 2010,
-            ConcurrentMerkleTreeError::InvalidProof(_, _) => 2011,
-            ConcurrentMerkleTreeError::CannotUpdateLeaf => 2012,
-            ConcurrentMerkleTreeError::CannotUpdateEmpty => 2013,
-            ConcurrentMerkleTreeError::AppendOnly => 2014,
-            ConcurrentMerkleTreeError::EmptyLeaves => 2015,
-            ConcurrentMerkleTreeError::EmptyChangelogEntries => 2016,
-            ConcurrentMerkleTreeError::MerklePathsEmptyNode => 2017,
-            ConcurrentMerkleTreeError::StructBufferSize(_, _) => 2018,
-            ConcurrentMerkleTreeError::FilledSubtreesBufferSize(_, _) => 2019,
-            ConcurrentMerkleTreeError::ChangelogBufferSize(_, _) => 2020,
-            ConcurrentMerkleTreeError::RootBufferSize(_, _) => 2021,
-            ConcurrentMerkleTreeError::CanopyBufferSize(_, _) => 2022,
+            ConcurrentMerkleTreeError::RootsZero => 2004,
+            ConcurrentMerkleTreeError::TreeFull => 2005,
+            ConcurrentMerkleTreeError::BatchGreaterThanChangelog(_, _) => 2006,
+            ConcurrentMerkleTreeError::InvalidProofLength(_, _) => 2007,
+            ConcurrentMerkleTreeError::InvalidProof(_, _) => 2008,
+            ConcurrentMerkleTreeError::CannotUpdateEmpty => 2009,
+            ConcurrentMerkleTreeError::EmptyLeaves => 2010,
+            ConcurrentMerkleTreeError::StructBufferSize(_, _) => 2011,
+            ConcurrentMerkleTreeError::FilledSubtreesBufferSize(_, _) => 2012,
+            ConcurrentMerkleTreeError::ChangelogBufferSize(_, _) => 2013,
+            ConcurrentMerkleTreeError::RootBufferSize(_, _) => 2014,
+            ConcurrentMerkleTreeError::CanopyBufferSize(_, _) => 2015,
```

Comment:
Seeing these error codes made me think, when do you think is a good opportunity to start adding the levels for error codes we talked about on a different pr?
(eg 7000ers Libraries, 8000ers Account Compression Program, etc.)

Reply:
we should think the actual error tiers through though haven't done that yet

Reply:
I think I could just file a PR with that now and rebase this one on top?

Reply:
Done here #698